src/full/Agda/Interaction/Base.hs

```diff
@@ -1,495 +1,25 @@
-{-# OPTIONS_GHC -fno-cse #-}
-{-# OPTIONS_GHC -fno-warn-orphans #-}
 
-module Agda.Interaction.Base where
+module Agda.Interaction.Base
+    ( module Agda.Interaction.Base
+    , module Agda.Interaction.Base.Boot
+    ) where
```

Comment:
It seems like almost the whole content of `Agda.Interaction.Base` got moved into `.Boot`.
I suggest to rename the new `A.I.Base.Boot` into `A.I.Base` after renaming `A.I.Base` to something suitable.  This would keep the diff much smaller, I suppose.
Also, having a `Boot` module with almost all of the content of the module it wants to boot is a bit out of the spirit of boot modules.

Reply:
Renamed to `.Types` in 8fecfbf, since it contains only type definitions, plus instances & functions needed for their `deriving` clauses.

Reply:
Maybe I didn't communicate this properly, but one of my request was to keep the code movements small.  Not sure if this is possible (since I didn't study the details), but it would be preferable.

Reply:
I think I see the problem now.

A second attempt is in 817f2c036e908498c8db8a7cb03dc75da55fc75e, which restores most of the contents of `A.I.Base`, and moves the TCM-aware functions to other modules (e.g. `A.I.InteractionTop`, `A.I.Output`)